addons/mail/populate/mail_channel.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.tools import populate
+
+
+class Channel(models.Model):
+    _inherit = "mail.channel"
+    _populate_dependencies = ["mail.message", "res.partner"]
+
+    _populate_sizes = {
+        'small': 100,
+        'medium': 1.105e5,
+        'large': 5.1e5,
+    }
+
+    def get_channel_probability(self):
+        return [0.1, 0.9]
+
+    def get_channel_type(self):
+        return ['channel', 'chat']
+
+    def _populate_factories(self):
+        message_ids = self.env.registry.populated_models['mail.message']
+        partner_ids = self.env.registry.populated_models['res.partner']
+
+        def get_partner_ids(values, counter, random):
+            if values.get('channel_type') == 'chat' and len(partner_ids) > counter:
+                partner = partner_ids[counter]
+                partners = [p for p in partner_ids if p != partner][:50]
+                return [(4, partner), (4, random.choice(partners))]
+            partners = random.choices(partner_ids, k=2)
+            return [(4, partners[0]), (4, partners[1])]
+
+        def get_channel_partner_ids(values, counter, random):
+            channel_partner_vals = []
+            for partner_id in values.get('channel_partner_ids'):
+                channel_partner_vals.append((0, 0, {
+                    'custom_channel_name': partner_id[1],
+                    'partner_id': partner_id[1]
+                }))
+            return channel_partner_vals
+
+        def get_messages(values, counter, random):
+            partner_ids = [val[1] for val in values['channel_partner_ids']]
+            message_records = self.env['mail.message'].browse(message_ids)
+            messages = message_records.filtered(lambda msg: not msg.res_id \
+                                                    and msg.model == 'mail.channel' \
+                                                    and msg.author_id.id in partner_ids)
+            return messages.ids[:50]
+
+        return [
+            ('channel_type', populate.randomize(self.get_channel_type(), self.get_channel_probability())),
+            ('channel_partner_ids', populate.compute(get_partner_ids)),
+            ('channel_last_seen_partner_ids', populate.compute(get_channel_partner_ids)),
+            ('message_ids', populate.compute(get_messages)),
+            ('name', populate.constant("channel - {counter}")),
+            ('public', populate.randomize(['public', 'private', 'groups'], [0.4, 0.4, 0.2])),
```

Comment:
For "groups" channel it would be expected to actually have groups linked.
And those groups should have users in them.